pyintegration/deephaven2/dtypes.py

```diff
@@ -24,174 +27,161 @@ def _qst_custom_type(cls_name: str):
+    """ Creates a Java array of the specified data type populated with values from a sequence.
+
+    Note:
+        this method does unsafe casting, meaning precision and values might be lost with down cast
+
+    Args:
+        dtype (DType): the component type of the array
+        seq (Sequence): a sequence of compatible data, e.g. list, tuple, numpy array, Pandas series, etc.
+        remap (optional): a callable that takes one value and maps it to another, for handling the translation of
+            special DH values such as NULL_INT, NAN_INT between Python and the DH engine
+
+    Returns:
+        a Java array
+
+    Raises:
+        DHError
+    """
+    try:
+        if remap:
+            if not callable(remap):
+                raise ValueError("Not a callable")
+            seq = [remap(v) for v in seq]
+        else:
+            if isinstance(seq, str) and dtype == char:
+                return array(char, seq, remap=ord)
+
+        return jpy.array(dtype.j_type, seq)
+    except Exception as e:
+        raise DHError(e, f"failed to create a Java {dtype.j_name} array.") from e
 
 
-class PropertiesDType(DType):
-    """ The class that wraps java Properties. """
+def from_jtype(j_class: Any) -> DType:
+    """ look up a DType that matches the java type, if not found, create a DType for it. """
+    if not j_class:
+        return None
 
-    def __call__(self, d: Dict) -> Any:
-        if d is None:
-            return None
-        r = self.j_type()
-        for key, value in d.items():
-            if value is None:
-                value = ''
-            r.setProperty(key, value)
-        return r
+    j_name = j_class.getName()
+    dtype = _j_name_type_map.get(j_name)
+    if not dtype:
+        return DType(j_name=j_name, j_type=j_class, np_type=np.object_)
+    else:
+        return dtype
 
 
-class HashMapDType(DType):
-    """ The class that wraps java HashMap. """
+def from_np_dtype(np_dtype: np.dtype) -> DType:
+    """ Look up a DType that matches the numpy.dtype, if not found, return PyObject. """
+    for _, dtype in _j_name_type_map.items():
+        if np.dtype(dtype.np_type) == np_dtype and dtype.np_type != np.object_:
+            return dtype
 
-    def __call__(self, d: Dict) -> Any:
-        if d is None:
-            return None
-        r = self.j_type()
-        for key, value in d.items():
-            if value is None:
-                value = ''
-            r.put(key, value)
-        return r
+    return PyObject
 
 
-class HashSetDType(DType):
-    """ The case the wraps java Set. """
+def is_java_type(obj: Any) -> bool:
+    """ Returns True if the object is originated in Java. """
+    return isinstance(obj, jpy.JType)
+
 
-    def __call__(self, s: Set):
-        if s is None:
-            return None
-        r = self.j_type()
-        for v in s:
-            r.add(v)
-        return r
+def j_array_list(values: Iterable):
```

Comment:
Should line 148 and below go in the types function, or should there be a separate module for it? e.g. `jcompat`.  None of it refers to anything else in `dtypes`.